### ISSUE TITLE
Error parsing table file

### ISSUE THREAD

**Issue by mcintyret** [mzgoddard/hard-source-webpack-plugin#366]
on `0.7.4` I hit this error immediately on the second build:

```
[hardsource:e6586c08] Tracking node dependencies with: package.json, node_modules/*.
[hardsource:e6586c08] Reading from cache e6586c08...
✖ ｢wdm｣: Error: Parse error on line 1:
...nd":593122},{"name":"normal-44136fa355b3
-----------------------^
Expecting 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '[', got 'undefined'
    at Object.parseError (/Users/.../node_modules/jsonlint/lib/jsonlint.js:55:11)
    at Object.parse (/Users/.../node_modules/jsonlint/lib/jsonlint.js:132:22)
    at Object.exports.parse (/Users/.../node_modules/jsonlint/lib/jsonlint.js:417:53)
    at module.exports.data (/Users/.../node_modules/hard-source-webpack-plugin/lib/util/parseJson.js:9:34)
    at Append2._readFile (/Users/.../node_modules/hard-source-webpack-plugin/lib/SerializerAppend2.js:164:19)
    at <anonymous>
```
other versions:
```
Mac OS 10.13.4
Node 8.9.4

  "devDependencies": {
    "copy-webpack-plugin": "^4.5.1",
    "hard-source-webpack-plugin": "^0.6.4",
    "jest": "^22.4.3",
    "npm-run-all": "^4.1.3",
    "speed-measure-webpack-plugin": "^1.2.2",
    "ts-jest": "^22.4.5",
    "tslint": "^5.10.0",
    "tslint-react": "^3.6.0",
    "typescript": "^2.8.3",
    "webpack": "^4.8.1",
    "webpack-cli": "^2.1.3",
    "webpack-dev-server": "^3.1.4"
  },
```

**Comment by Michael-Tajmajer-Emrsn** [comment]
I'm seeing the same issue, second run results in this error:

[hardsource:ba1afca6] Tracking node dependencies with: yarn.lock, package.json, webpack.config.ts.
[hardsource:ba1afca6] Reading from cache ba1afca6...
Error: Parse error on line 1:
...nd":210727},{"name":"normal-44136fa355b3
-----------------------^
Expecting 'STRING', 'NUMBER', 'NULL', 'TRUE', 'FALSE', '{', '[', got 'undefined'
    at Object.parseError ([project path]node_modules\jsonlint\lib\jsonlint.js:55:11)
    at Object.parse ([project path]node_modules\jsonlint\lib\jsonlint.js:132:22)
    at Object.exports.parse ([project path]node_modules\jsonlint\lib\jsonlint.js:417:53)
    at module.exports.data ([project path]node_modules\hard-source-webpack-plugin\lib\util\parseJson.js:9:34)
    at Append2._readFile ([project path]node_modules\hard-source-webpack-plugin\lib\SerializerAppend2.js:164:19)
    at <anonymous>

Windows 10, Node 8.11.2

Packages are all pretty fresh -- a lot more than mcintyret listed :-)

**Comment by Michael-Tajmajer-Emrsn** [comment]
Note: this doesn't happen all the time -- if I delete the .hard_cache folder and rebuild it will sometimes work.

**Comment by mcintyret** [comment]
It looks like some of the cache files are currupted. In the middle of the 2.3Mb file I see this:

`{"name":"normal-44136fa3["../../node_modules/webpack/hot"]["../../node_modules/moment/locale"]...`

So the json is indeed invalid.

That cutoff is at the byte offset specified by `tableLength`, so maybe there's a bug in the offset calculation when the files are being written?

**Comment by mcintyret** [comment]
Looks like it was happening because the buffers weren't being expanded properly at write time. Fix here: https://github.com/mzgoddard/hard-source-webpack-plugin/pull/369

**Comment by mzgoddard** [comment]
@mcintyret Thank you for finding that bug in Append2. I merged your change in #374 and published as `0.7.5`.
